### PR TITLE
Site Assembler - Rename category Featured as All to explore all patterns

### PR DESCRIPTION
**Pull request by miksansegundo**
Related to https://github.com/Automattic/wp-calypso/issues/77129 #76451 #76580 

## Proposed Changes

* Rename the category `Featured` as `All` in the assembler only 
* Update the description
* Enable feature flag in staging and production

This is not in the editor because there is a performance issue when rendering a category with thousands of patterns. See https://github.com/Automattic/wp-calypso/issues/77129

Screenshot 2566-05-19 at 18 58 18

## Testing Instructions

- Create a new site with any plan
- Continue until the Design Picker
- Scroll down and click `Start designing`
- Click `Homepage` > `Add patterns` 
- Verify that the category `All` is the first and contains all patterns
- Verify that there aren't performance issues when scrolling all the patterns

## Pre-merge Checklist

- [X] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [X] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [X] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [X] Have you checked for TypeScript, React or other console errors?
- [X] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [X] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [X] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

**Review by arthur791004 — COMMENTED** [review]
Since we just rename the label to `All`, how about doing it when rendering the category?

**Comment by miksansegundo** [comment]
> Since we just rename the label to `All`, how about doing it when rendering the category?

I think because we need the category title and description in the panel too for this https://github.com/Automattic/wp-calypso/pull/77087

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx [hunk 1]
```diff
@@ -84,7 +85,8 @@ const PatternAssembler = ( {
 		() => allPatterns.map( ( pattern ) => encodePatternId( pattern.ID ) ),
 		[ allPatterns ]
 	);
-	const categories = usePatternCategories( site?.ID );
+	const theCategories = usePatternCategories( site?.ID );
+	const categories = useAddCategoryAll( theCategories );
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx]: I guess we can do it inside `usePatternCategories`. Or using the `select` functionality of the react-query to replace the label and description. But it's not a blocker, I'm fine with the current way 🙂

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx [hunk 2]
```diff
@@ -84,7 +85,8 @@ const PatternAssembler = ( {
 		() => allPatterns.map( ( pattern ) => encodePatternId( pattern.ID ) ),
 		[ allPatterns ]
 	);
-	const categories = usePatternCategories( site?.ID );
+	const theCategories = usePatternCategories( site?.ID );
```
Comment (arthur791004) [inline client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx]: Naming is difficult 😓

**Review by arthur791004 — APPROVED** [review]

File: client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx (hunk 1 above)
Comment (miksansegundo) [reply client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx]: I updated the code to use the `select` in the query. I think is better. WDYT?

**Review by arthur791004 — APPROVED** [review]
Still works well 🙂

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/8006340

Thank you @miksansegundo for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/8006340).

**Pull request merged**